disnake/ext/commands/core.py

```diff
@@ -97,25 +97,30 @@
 
 MISSING: Any = disnake.utils.MISSING
 
-T = TypeVar("T")
 VT = TypeVar("VT")
-CogT = TypeVar("CogT", bound="Optional[Cog]")
 CommandT = TypeVar("CommandT", bound="Command")
-ContextT = TypeVar("ContextT", bound="Context")
 GroupT = TypeVar("GroupT", bound="Group")
 HookT = TypeVar("HookT", bound="Hook")
 ErrorT = TypeVar("ErrorT", bound="Error")
 
 
 if TYPE_CHECKING:
-    P = ParamSpec("P")
+    from typing_extensions import TypeVar  # noqa: TC004
 
+    P = ParamSpec("P", default=...)
+    T = TypeVar("T", default=Any)
+
+    CogT = TypeVar("CogT", bound="Optional[Cog]", default="Optional[Cog]")
+    ContextT = TypeVar("ContextT", bound="Context", default="Context")
     CommandCallback = Union[
         Callable[Concatenate[CogT, ContextT, P], Coro[T]],
         Callable[Concatenate[ContextT, P], Coro[T]],
     ]
 else:
+    T = TypeVar("T")
     P = TypeVar("P")
+    CogT = TypeVar("CogT", bound="Optional[Cog]")
+    ContextT = TypeVar("ContextT", bound="Context")
```

Comment:
https://github.com/DisnakeDev/disnake/pull/1441/files#r2443495659

```suggestion
    CogT = TypeVar("CogT")
    ContextT = TypeVar("ContextT")
```